src/lib/engine/space.js

```diff
@@ -47,13 +48,15 @@ export function toDSpace(spaceId, graph) {
 
   dSpace.edges = _dGraph.dependencyMap(dSpace)
 
+  const withInputFn = _guesstimate.inputsFromExpressions(dSpace.metrics)
   dSpace.metrics = dSpace.metrics.map(s => {
     let edges = {}
     edges.inputs = dSpace.edges.filter(i => i.output === s.id).map(e => e.input)
     edges.outputs = dSpace.edges.filter(i => i.input === s.id).map(e => e.output)
     edges.inputMetrics = edges.inputs.map(i => dSpace.metrics.find(m => m.id === i))
-    return Object.assign({}, s, {edges})
+    return {...s, guesstimate: withInputFn(s.guesstimate), edges}
```

Comment:
for readability, I think edges should be second in this,  Also, you may want to split it for multiple lines.

```
return {
  ...s, 
  edges,
  guesstimate: withInputFn(s.guesstimate)
}
```